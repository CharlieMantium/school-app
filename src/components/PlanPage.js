import React from 'react';
import ActivitiesList from './ActivitiesList';
import ActivitiesListFilter from './ActivitiesListFilter';

const PlanPage = () => (
  <div>
    <ActivitiesListFilter />
    <ActivitiesList />
  </div>
);

export default PlanPage;
