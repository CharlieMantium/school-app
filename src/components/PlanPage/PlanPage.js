import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import ActivitiesListFilter from '../ActivitiesListFilter';
import Week from '../Week';
import { startSetActivities } from '../../store/activities/actions';

class PlanPage extends React.Component {
  static propTypes = {
    onStartSetActivities: PropTypes.func,
  };

  static defaultProps = {
    onStartSetActivities: () => {},
  };

  componentDidMount() {
    this.props.onStartSetActivities();
  }

  render() {
    return (
      <div>
        <ActivitiesListFilter data-test="filter-component" />
        <Week data-test="week-component" />
      </div>
    );
  }
}

const mapDispatchToProps = {
  onStartSetActivities: startSetActivities,
};

export { PlanPage as PlanPageUnwrapped };
export default connect(
  null,
  mapDispatchToProps,
)(PlanPage);
