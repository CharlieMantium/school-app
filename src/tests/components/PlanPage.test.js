import React from 'react';
import { shallow } from 'enzyme';

import PlanPage from '../../components/PlanPage';

describe('PlanPage', () => {
  it('should render PlanPage correctly', () => {
    const wrapper = shallow(<PlanPage />);
    expect(wrapper).toMatchSnapshot();
  });
});
