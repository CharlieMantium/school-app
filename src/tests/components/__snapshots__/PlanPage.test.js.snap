// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanPage should render PlanPage correctly 1`] = `
<div>
  <ActivitiesListFilter />
  <Connect(Week) />
</div>
`;
